### PR TITLE
Table: Row style overrides

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Providing new prop for parent component of table to override row styles.

image

This PR shouldn't change anything, just provides more options to developers building UIs on top of table

### CODE REVIEW

**Review by torkelo — CHANGES_REQUESTED** [review]
This is sadly not the right direction I think, needs to be handled via field config and be data driven (so you can use it with thresholds or color scales)  

We have cell styles that color cells. We need an option on the colored cell to make it apply to the full row.

**Comment by torkelo** [comment]
At least that is how I would try to solve this, not sure how easy / possible it is :)

File: packages/grafana-ui/src/components/Table/RowsList.tsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ interface RowsListProps {
   timeRange?: TimeRange;
   footerPaginationEnabled: boolean;
   initialRowIndex?: number;
+  rowStyles?: Map<number, React.CSSProperties>;
```
Comment (matyax) [inline packages/grafana-ui/src/components/Table/RowsList.tsx:48]: I like this, very clever.

**Review by matyax — APPROVED** [review]
👌👌

**Comment by matyax** [comment]
Oh apologies, I missed the previous review. This approach makes sense for our use case, but let's explore other options, if there are.

**Comment by gtk-grafana** [comment]
> This is sadly not the right direction I think, needs to be handled via field config and be data driven (so you can use it with thresholds or color scales)
> 
> We have cell styles that color cells. We need an option on the colored cell to make it apply to the full row.

@torkelo I chatted with @codeincarnate yesterday, and my understanding of the discussion was that the current proposed solution that will be exposed to end users in the table options here: https://github.com/grafana/grafana/pull/82203, would likely not be ready in Q1 for o11y-logs to hit our MVP deadline, at least not working with string enum values (our use-case is to visualize the log level using the background color of rows), so Kyle suggested that we propose a solution that works for o11y-logs in the short term while dataviz takes the time to do it right, with the plan that we'd clean this up and migrate to the longer term solution when it's available. 

Would love to hear your thoughts on this and thank you for the feedback!

**Comment by torkelo** [comment]
@gtk-grafana not sure why we need two solutions? that seems to just add so much complexity. Reusing our exisiting value and coloring options seems to me to be the hugly easier and less complex option that should be faster to implement and get into mergable state.

quick poc 
https://github.com/grafana/grafana/pull/83247

**Comment by gtk-grafana** [comment]
Awesome, thank you! I'll take a look at that PoC but at first glance this looks like it would work great.

**Comment by gtk-grafana** [comment]
https://github.com/grafana/grafana/pull/83247 appears to work and looks like a better implementation for the future of the component. closing!

**Pull request closed without merging**